src/genericitems/list.cpp

```diff
@@ -2,293 +2,186 @@
-            {
-                // If we have already checked this item, move on...
-                if (checkedItems.contains(item))
-                    continue;
-
-                // Found an item that hasn't already been checked, so flag to the slaves that we will be sending
-                // another name to check
-                if (!procPool.decideTrue(n))
-                    return false;
+    auto it =
+        std::find_if(classNames_.begin(), classNames_.end(), [&className](auto &item) { return item.second == className; });
+    if (it == classNames_.end())
+        throw(std::runtime_error(
+            fmt::format("Item '{}' cannot be created as no producer has been registered for class '{}'.\n", name, className)));
 
-                Messenger::printVerbose("Checking equality of item '{}'...\n", item->name());
+    items_.emplace(name, GenericItem(producers_[it->first](), className, version, flags));
 
-                // Send name of item - slaves will try to find it, so we need to check that everybody does...
-                itemName = item->name();
-                if (!procPool.broadcast(itemName, n))
-                    return false;
-                if (!procPool.allTrue(true))
-                {
-                    Messenger::error("GenericList equality check failed - item '{}' is not present on all "
-                                     "processes.\n",
-                                     item->name());
-                    ++nFailed;
-                    continue;
-                }
-
-                // Send class name of item - slaves will check it, and everybody must agree
-                itemClassName = item->itemClassName();
-                if (!procPool.broadcast(itemClassName, n))
-                    return false;
-                if (!procPool.allTrue(true))
-                {
-                    Messenger::error("GenericList equality check failed - item '{}' is not of the same "
-                                     "type ({}) on all processes.\n",
-                                     item->name(), item->itemClassName());
-                    ++nFailed;
-                    continue;
-                }
+    return items_[std::string(name)];
+}
 
-                // Check the item for equality
-                if (!item->equality(procPool))
-                    return Messenger::error("Failed equality check for module data '{}'.\n", item->name());
+/*
+ * Serialisation
+ */
 
-                // Add the item to our checked list
-                checkedItems.append(item);
-            }
+// Serialise all objects via the specified LineParser
+bool GenericList::serialiseAll(LineParser &parser, const CoreData &coreData, std::string_view headerPrefix) const
+{
+    for (auto &item : items_)
+    {
+        // If it is not flagged to be saved in the restart file, skip it
+        if (!(std::get<ItemData::Flags>(item.second) & GenericList::InRestartFileFlag))
+            continue;
```

Comment:
```suggestion
    for (auto &[key, value] : items_)
    {
        // If it is not flagged to be saved in the restart file, skip it
        if (!(std::get<ItemData::Flags>(value) & GenericList::InRestartFileFlag))
            continue;
```

It's a style thing, but I happen to like using the destructuring binds to give names to the individual parts

Reply:
Much nicer!